File: docs/snippets/src/csharp/handling-input.cs
float speed = 200f;

while (!game.ShouldClose())
{
    game.BeginFrame(0.1f, 0.1f, 0.1f, 1.0f);

    float dt = game.DeltaTime;

    if (game.IsKeyPressed(Keys.Escape))
    {
        game.Close();
        continue;
    }

    if (game.IsKeyPressed(Keys.Left))  x -= speed * dt;
    if (game.IsKeyPressed(Keys.Right)) x += speed * dt;
    if (game.IsKeyPressed(Keys.Up))    y -= speed * dt;
    if (game.IsKeyPressed(Keys.Down))  y += speed * dt;
    if (game.IsKeyPressed(Keys.Space)) { }

    game.DrawSprite(textureId, x, y, width, height);

    game.EndFrame();
}

File: docs/snippets/src/typescript/handling-input.ts
while (!game.shouldClose()) {
  game.beginFrame(0.1, 0.1, 0.1, 1.0);

  if (game.isKeyPressed(256)) {
    break;
  }
  if (game.isKeyPressed(32) || game.isMouseButtonPressed(0)) {
  }
  if (game.isKeyPressed(263)) {
  }
  if (game.isKeyPressed(262)) {
  }

  game.endFrame();
}
